Formula/jrnl.rb

```diff
@@ -16,79 +15,167 @@ class Jrnl < Formula
+    sha256 "236bdbdce46e6e6a3d61a337c0f8b763ca1e8717c03b369e87a7ec7ce1319c0a"
   end
 
   resource "tzlocal" do
     url "https://files.pythonhosted.org/packages/cb/89/e3687d3ed99bc882793f82634e9824e62499fdfdc4b1ae39e211c5b05017/tzlocal-1.5.1.tar.gz"
     sha256 "4ebeb848845ac898da6519b9b31879cf13b6626f7184c496037b818e238f2c4e"
   end
 
+  resource "zipp" do
+    url "https://files.pythonhosted.org/packages/11/b5/89f3ab6d45b2709863761bab58c574b2344ef215749abb5407818c21c9ca/zipp-2.1.0.tar.gz"
+    sha256 "feae2f18633c32fc71f2de629bfb3bd3c9325cd4419642b1f1da42ee488d9b98"
+  end
+
   def install
     virtualenv_install_with_resources
   end
 
   test do
-    (testpath/"write_journal.sh").write <<~EOS
+    (testpath/"write_journal.sh").write <<~'EOS'
+      #!/usr/bin/expect -f
+
+      set timeout 3
+      match_max 100000
+      spawn "$env(BREW_TEST_BIN)/jrnl" "This is the fanciest test in the world."
+
+      expect {
+        "Path to your journal file" { send -- "$env(BREW_TEST_PATH)/test.txt\r" }
+        timeout { exit 1 }
+      }
+
+      expect {
+        -exact "Do you want to encrypt your journal? You can always change this later \[y/N\] " { send -- "n\r" }
+        timeout { exit 1 }
+      }
+
+      expect {
+        eof { exit }
+        timeout { exit 1 }
+      }
+    EOS
+
+    (testpath/"read_journal.sh").write <<~'EOS'
       #!/usr/bin/expect -f
-      set timeout -1
-      spawn #{bin}/jrnl today: Wrote this fancy test.
-      expect -exact "Path to your journal file (leave blank for ~/journal.txt):"
-      send -- "#{testpath}/journal\n"
-      expect -exact "Enter password for journal (leave blank for no encryption): "
-      send -- "Homebrew\n"
-      expect "Do you want to store the password in your keychain?"
-      send -- "N\n"
-      expect -exact "Journal will be encrypted."
-      expect "Entry added to default journal"
-      expect eof
+
+      set timeout 3
+      spawn "$env(BREW_TEST_BIN)/jrnl" -1
+
+      expect {
+        "This is the fanciest test in the world." { exit }
+        timeout { exit 1 }
+        eof { exit 1 }
+      }
+
+    EOS
+
+    (testpath/"encrypt_journal.sh").write <<~'EOS'
+      #!/usr/bin/expect -f
+
+      set timeout 3
+      spawn "$env(BREW_TEST_BIN)/jrnl" --encrypt
+
+      expect {
+        -exact "Enter password for new journal: " { send -- "homebrew\r" }
+        timeout { exit 1 }
+      }
+
+      expect {
+        -exact "Enter password again: " { send -- "homebrew\r" }
+        timeout { exit 1 }
+      }
+
+      expect {
+        -exact "Do you want to store the password in your keychain? \[Y/n\] " { send -- "n\r" }
+        timeout { exit 1 }
+      }
+
+      expect {
+        -re "Journal encrypted to .*/test.txt." { exit }
+        timeout { exit 1 }
+        eof { exit 1 }
+      }
+    EOS
+
+    (testpath/"test_journal.sh").write <<~EOS
```

Comment:
Please do all of this just in ruby below. The current approach makes it difficult to debug where it failed. Preferably there would be as few scripts as possible.

Reply:
Hi, I've updated the PR with your feedback. I've opted to keep using expect since our project has much more knowledge of cli tools than Ruby, so this will be easier to maintain.

Reply:
It's a homebrew formula though, you're not a maintainer of homebrew. I'd prefer it if it were easier for homebrew maintainers and contributors to work with.